app/tests/V1/test.py

```diff
@@ -257,6 +257,268 @@ def test_invalid_weight(self):
+        self.assertEqual(res.status_code, 200)
+
+    def test_mark_order_as_moving(self):
+        """Test mark order as moving."""
+        res = self.client.put(
+            "api/v2/parcels/1/moving",
+
+            headers={"content-type": "application/json"}
+        )
+
+        self.assertEqual(res.status_code, 404)
+        self.assertEqual(json.loads(res.data)[
+                         'message'], "order not found")
+
+    def test_decline_an_order(self):
+        """Test admin decline order."""
+        res = self.client.put(
+            "api/v2/parcels/1/declined", data=json.dumps(self.data),
+            headers={"content-type": "application/json"}
+        )
+        self.assertEqual(res.status_code, 200)
+
+    def test_invalid_origin_name(self):
+        """Test invalid origin name."""
+        data = {
+            "origin": "******",
+            "price": 20,
+            "destination": "Nairobi",
+            "weight": 5
+        }
+
+        res = self.client.post(
+            "api/v2/parcels",
+            data=json.dumps(data),
+            headers={"content-type": "application/json"}
+        )
+
+        self.assertEqual(res.status_code, 400)
+        self.assertEqual(json.loads(res.data)[
+                         'message'], "invalid origin name")
+
+    def test_invalid_destination(self):
+        """Test invalid destination."""
+        data = {
+            "origin": "nairobi",
+            "price": 20,
+            "destination": "!!!!!!!",
+            "weight": 5
+        }
+
+        res = self.client.post(
+            "api/v2/parcels",
+            data=json.dumps(data),
+            headers={"content-type": "application/json"}
+        )
+
+        self.assertEqual(res.status_code, 400)
+        self.assertEqual(json.loads(res.data)[
+                         'message'], "destination not valid")
+
+    def test_invalid_price(self):
+        """Test invlid price."""
+        data = {
+            "origin": "kiambu",
+            "price": "kevo",
+            "destination": "kiambu",
+            "weight": 5
+        }
+
+        res = self.client.post(
+            "api/v2/parcels",
+            data=json.dumps(data),
+            headers={"content-type": "application/json"}
+        )
+
+        self.assertEqual(res.status_code, 400)
+        self.assertEqual(json.loads(res.data)['message'], {'price': 'Invalid price'})
+
+    def test_invalid_weight(self):
+        """Test invalid weight."""
+        data = {
+            "origin": "kiambu",
+            "price": 50,
+            "destination": "kiambu",
+            "weight": "vdhdsfd"
+        }
+
+        res = self.client.post(
+            "api/v2/parcels",
+            data=json.dumps(data),
+            headers={"content-type": "application/json"}
+        )
+
+        self.assertEqual(res.status_code, 400)
+        self.assertEqual(json.loads(res.data)['message'], {'weight': 'Invalid weight'})
+
+
+if __name__ == '__main__':
```

Comment:
Kindly remove one of this blocks of codes.

Reply:
I removed one of them, it was a copy/paste error.